doc/release/upcoming_changes/18637.improvement.rst

```diff
@@ -0,0 +1,3 @@
+``count_nonzero`` has been added to ``np.ma``
+----------------------------------------------
+When ``np.count_nonzero`` is applied to a masked array, it considers masked zero value as a non-zero value. Now, ``np.ma.count_nonzero`` has been implemented to avoid this issue by separating the array from the mask. The function returns the number of non-zero unmasked values in a given array. To use it, call ``np.ma.count_nonzero(arr, options)``. 
```

Comment:
Please keep the lines < 80 characters in length.

Reply:
on it!

Reply:
Sorry to be late, I think the doc is misleading: ``np.count_nonzero`` applied to a masked array *does not* consider masked zero value as non-zero values, but rather does inappropriately consider masked non-zero values. I would rather say simply:

``np.count_nonzero`` applied to a masked array does not consider mask; ``np.ma.count_nonzero`` has been implemented to avoid this issue.